lighthouse-core/lib/url-shim.js

```diff
@@ -96,6 +96,19 @@ URL.getURLDisplayName = function getURLDisplayName(url, options) {
   return Util.getURLDisplayName(new URL(url), options);
 };
 
+/**
+ * @param {string} url
+ * @return {string}
+ */
+URL.elideDataURI = function elideDataURI(url) {
+  try {
+    const parsed = new URL(url);
+    return parsed.protocol === 'data:' ? url.slice(0, 100) : url;
```

Comment:
> it does mean that any canonicalization or whatever that the URL constructor does is included

I'm a little confused why we want this, it's going to be reprocessed by a `new URL` by the renderer in `_renderUrl` anyhow, so why would we want to mess with it more here?